skypy/utils/random.py

```diff
@@ -57,21 +56,19 @@ def schechter(alpha, x_min, x_max, resolution=100, size=None, scale=1.):
     if size is None:
         size = np.broadcast(x_min, x_max, scale).shape or None
 
-    x = np.logspace(np.log10(np.min(x_min)), np.log10(np.max(x_max)),
-                    resolution)
-    cdf = _schechter_cdf(x, np.min(x_min), np.max(x_max), alpha)
-    t_lower = np.interp(x_min, x, cdf)
-    t_upper = np.interp(x_max, x, cdf)
-    u = np.random.uniform(t_lower, t_upper, size=size)
-    x_sample = np.interp(u, cdf, x)
+    lnx_min = np.log(x_min)
+    lnx_max = np.log(x_max)
 
-    return x_sample * scale
+    lnx = np.linspace(np.min(lnx_min), np.max(lnx_max), resolution)
 
+    cdf = np.exp(np.add(alpha, 1)*lnx - np.exp(lnx))
+    np.cumsum((cdf[1:]+cdf[:-1])/2*np.diff(lnx), out=cdf[1:])
+    cdf[0] = 0
```

Comment:
```suggestion
    pdf = np.exp(np.add(alpha, 1)*lnx - np.exp(lnx))
    cdf = np.zeros_like(pdf)
    cdf[1:] =  np.cumsum((cdf[1:]+cdf[:-1])/2*np.diff(lnx))
```
Although this suggestion uses an extra variable I find it much clearer. Don't mind either way

Reply:
```suggestion
    pdf = np.exp(np.add(alpha, 1)*lnx - np.exp(lnx))
    cdf = pdf  # in place
    np.cumsum((pdf[1:]+pdf[:-1])/2*np.diff(lnx), out=cdf[1:])
    cdf[0] = 0
```

This would still perform the operation in place if it's only about the variable name.